docs/stdlib/cfg.rst

```diff
@@ -116,15 +116,26 @@ Client connections
 :eql:synopsis:`allow_user_specified_id -> bool`
   Makes it possible to set the ``.id`` property when inserting new objects.
 
-  Enabling this feature introduces some security vulnerabilities:
+  .. warning::
 
-  1. An unprivileged user can discover ids that already exist in the database
-     by trying to insert new values and noting when there is a constraint
-     violation on ``.id`` even if the user doesn't have access to the relevant
-     table.
+      Enabling this feature introduces some security vulnerabilities:
 
-  2. It allows re-using object ids for a different object type, which the
-     application might not expect.
+      1. An unprivileged user can discover ids that already exist in the
+         database by trying to insert new values and noting when there is a
+         constraint violation on ``.id`` even if the user doesn't have access
+         to the relevant table.
+
+      2. It allows re-using object ids for a different object type, which the
+         application might not expect.
+
+      Additionally, enabling can have serious performance implications as, on
```

Comment:
I would say "every object type must be checked for collisions" instead. It's kind of a fine distinction, but "scanned" implies worse than it actually is, because my reading of it is that it would actually need to check every object, which it shouldn't: it will do a separate index lookup for each type.